### ISSUE TITLE
Dimension Support

### ISSUE THREAD

**Issue by Borkmeir** [BrainSlugs83/NewBiospheresMod#31]
Just in case, I read through the Git a bit and the MCF post as well, but I didn't find anything so I'll just ask.

Do you have any plans to add dimension support to this?

What I was thinking was the option to assign Biospheres to different dimensions, while leaving the Overworld intact.  Ideally, the ability to configure, maybe, up to five separate Biosphere dimensions.  I suppose this would also require some method of dimensional transport as well.  It could be interesting to base a modpack around the idea of jumping from dimension to dimension to get various materials.  Some players would probably enjoy living in such a dimension on my server, but some would not.  So at the very least being able to set the dimension of the biospheres would be interesting.

**Comment by TheLoneWolfling** [comment]
This is something that could be added relatively easily. The internals are already set up around a dimension provider, it's just adding the boilerplate to allow specific dimensions instead of just overwriting the overworld.

Actual teleportation may be outside the scope of this mod, however.

Though maybe just overriding /tp to allow teleportation to a different dimension... Would that work? You can always plunk down command block teleportation stations.

**Comment by Borkmeir** [comment]
Oh I have no issues with teleportation, as I have a number of mods that offer the functionality I need.  I was just thinking that if someone was playing without those mods and wanted to still have a normal Overworld they might desire an inbuilt mechanic in the mod.

**Comment by BrainSlugs83** [comment]
I think this came up in the original thread didn't it? -- Someone was saying there's another mod that lets you manage this easily (assign different world generators to different dimensions) -- I think there's a bug with this mod though -- something was holding them back maybe(?)... let me see.

**Comment by BrainSlugs83** [comment]
It seems maybe it's called Multi World? -- I'm not sure if it works or not.  If we need to change something, let me know.

The only part where we're explicitly working against dimension 0 is for the respawn code -- everything else is based on the original mod, which -- as far as I can find it -- doesn't specify any dimension (even at registration time).

**Comment by Borkmeir** [comment]
The first Multi World mod I found seems to only have support for specific generators according to its description.  It's also been 10 months since its last update, so it may be dead now.  The other one is a part of Forge Essentials, which is a well intentioned mod that has never worked correctly, even once, in my experience, and has conflicts with at least 4 mods in my pack.

Unfortunately, in my specific case, even if there is a Forge mod that can make this a reality for me, I would prefer not to add two mods just to get the functionality of one, as the test version of my pack already has 220+ mods in it, though that number will drop below 200 when I get around to pruning it.  Ultimately, it will probably be around 150 mods + client side only mods, and get split into Tech / Magic servers, as even 150 mods can be asking for trouble on a server.

I'm not trying to be pushy about this.  All I really want is to be able to assign your mod to a specific dimension.  I wouldn't need anything more for my purposes.  If you say no, I'll understand.  If it's outside the scope of what you wish to do, then no big deal.  I just thought it would be neat to offer my players yet another unique place to live next to the other 12 or so dimensions my pack already offers, but I'm not wedded to the idea.

**Comment by TheLoneWolfling** [comment]
I can do up a version with this. Gimme a few hours, I've got a couple other things to do.

The bigger problem is making sure it remains compatible in the future.

**Comment by BrainSlugs83** [comment]
Yeah, apologies -- to be clear -- I'm not trying to push back either. -- It's a good suggestion, I was just hoping there was a way to get you want you want both sooner for you, and with less effort for me -- and it sounds like -- so far, there's not a great way to do it. -- So the mod will have to do it.

Currently, though, I don't think the mod is specifying a dimension explicitly (I think it's getting 0 by default somehow), so that means we'll have to do some extra coding to make it work. -- @TheLoneWolfling do you have any experience with this?

Or @Borkmeir, is there an existing mod that does this well (in your opinion?) -- that is, with Forge 1.7.10? -- I might try to peek at it and see what it's doing and try to adapt that to this mod. -- I'm glad you logged the issue, 'cause that sort of fell off the back burner and got forgotten about, and now it's tracked!

**Comment by Borkmeir** [comment]
I'll look around, and see what's out there.  However, if you're looking for mods that create their own dimension for their terrain generator, I would think that Twilight Forest, ATUM, or even Aether II all implement something similar to what you would need.  Though of course, all those mods do more than just create a new dimension with different generation.

**Comment by BrainSlugs83** [comment]
Cool, I'll take a look at one of those.
